kubernetes/kubernetes.go

```diff
@@ -136,6 +143,7 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 	kubeAgent := newKubeAgent(ctx, config)
 
 	customS3Mode := isCustomS3UploadEnvsSet(&kubeAgent)
+	customAzureMode := isCustomAzureUploadEnvsSet(&kubeAgent)
```

Comment:
Now that we have two modes, I wonder if it makes sense to make an `agentUploadMode` variable instead of managing separate vars and passing double parameters. Especially if other locations are needed to be supported in the future GKE, OKE...
Basically could have a map of `uploadModes` that could determine future actions